functions/Invoke-DbaSqlQuery.ps1

```diff
@@ -233,7 +240,43 @@ function Invoke-DbaSqlQuery {
 
     process {
         if (Test-FunctionInterrupt) { return }
+        foreach($el in $DatabaseCollectionLoose) {
+            if ($el.SQLInstance -and $el.Database) {
+                $DatabaseCollection = $el.Database
+            }
+        }
+        if ((Test-Bound -ParameterName "Database") -and ((Test-Bound -ParameterName "DatabaseCollection") -or $DatabaseCollection)) {
+            Stop-Function -Category InvalidArgument -Message "You can't use -Database with piped databases"
+            return
+        }
+        if ((Test-Bound -ParameterName "SqlInstance") -and (Test-Bound -ParameterName "DatabaseCollection")) {
+            Stop-Function -Category InvalidArgument -Message "You can't use -SqlInstance with piped databases"
+            return
```

Comment:
Wouldn't parametersets be more helpful here?
Then the legal combinations will show up in help...
On the other hand, that would conflict with DatabaseCollectionLoose ...